File: ios/Cookpit/Cookpit/HttpClient.swift
//
//  Networking.swift
//  Cookpit
//
//  Created by Kittinun Vantasin on 5/31/16.
//  Copyright © 2016 Cookpit. All rights reserved.
//

import Foundation
import Alamofire

class HttpClient {
}

extension HttpClient : CPHttp {
  @objc func get(url: String, params: [String : String]?, observer: CPHttpObserver?) {
      var modifiedParams = params
  
      if modifiedParams != nil {
        modifiedParams!["format"] = "json"
        modifiedParams!["nojsoncallback"] = "1"
      }
  
      Alamofire.request(.GET, url, parameters: modifiedParams).validate().responseString { response in
        switch response.result {
          case .Success(let data) :
            observer?.onSuccess(data)
          case .Failure :
            observer?.onFailure()
        }
      }
  }
}

File: android/Cookpit/app/src/main/kotlin/com/github/kittinunf/cookpit/networking/HttpClient.kt
package com.github.kittinunf.cookpit.networking

import com.github.kittinunf.cookpit.Http
import com.github.kittinunf.cookpit.HttpObserver
import com.github.kittinunf.fuel.Fuel
import com.github.kittinunf.result.Result

class HttpClient : Http() {

    override fun get(url: String, observer: HttpObserver?) {
        Fuel.get(url).responseString{ request, response, result ->
            when (result) {
                is Result.Success -> observer?.onSuccess(result.value)
                is Result.Failure -> observer?.onFailure()
            }
        }
    }

}